fix logreport

anb76ru opened the pull request:
Hello!  I use the `pytest_runtest_logreport` method to get subtest reports.  A report is generated for `when='call'`, but if the test failed, the `report` is not generated. I think this is because the `log_passing_tests=True` attribute is used in the `elif report.failed:` block. If the test failed, then if `when == 'call'` we don't step into the code block:
```python
if not self.log_passing_tests:        
    reporter.write_captured_output(report)
```
and the report is not generated. I don't understand why  `self.log_passing_test using` in `report.failed`.  Maybe this is a mistake.  If you remove the '```if not self.log_passing_tests: ```', then the report for the failed tests for `when='call'` is generated without problems.

I suggest making `reporter.write_captured_output(report)` unconditional

#https://github.com/pytest-dev/pytest/issues/10491

OS: win10
pytest_ver:  #7.2.0
![log_passing_tests](https://user-images.githubusercontent.com/65860480/201186826-45eaba3a-2d7c-4363-bfad-a894d72b47b8.jpg)

nicoddemus reviewed (changes requested):
Thanks! The fix does make sense.

I also noticed that `write_captured_output` should be called in the start of the function, in the `if report.passed`/`if report.when == "call"`, otherwise passing tests won't output the captured output even if `log_passing_tests` is True.

We also need tests and a changelog entry. 👍

anb76ru commented:
@nicoddemus , if ```write_captured_output``` call without condition ```if not self.log_passing_tests```, than
get failed test ```test_avoid_double_stdout```. 
![result_test_avoid_double_stdout](https://user-images.githubusercontent.com/65860480/202016509-68beaf20-5076-48ca-96cc-bb4225a571a5.jpg)

Therefore, I moved the condition ```if not self.log_passing_tests``` in ```if report.passed``` / ```if report.when == "call"```. But I am not sure about the correctness of these changes
Now, all test passed
Also, I added information about changes in file changelog.

Please, check PR

anb76ru commented:
@nicoddemus, can you review, please ?

anb76ru commented:
@nicoddemus I remove ```condition if not self.log_passing_tests``` in ```if report.passed / if report.when == "call"```
add test, and change target branch to main. Review please

anb76ru commented:
@nicoddemus, Hello. Please check my changes

nicoddemus commented:
Closing as per https://github.com/pytest-dev/pytest/issues/10491#issuecomment-1340914543.

Sorry again for the misunderstanding of the use case, and thanks for the effort @anb76ru!
